RELEASE.md

```diff
@@ -0,0 +1,106 @@
+# Release process
+
+This file documents the necessary steps for releasing Irmin to its various users
+(via GitHub, `opam-repository` and Tezos).
+
+At a high level, releasing Irmin consists of publishing the following artefacts:
+
+- a Git [commit tag][git-tags];
+- a set of documentation on GitHub pages (e.g. [`mirage.github.io/irmin`][pages-docs]);
+- a release archive (`.tbz` file containing the project source) on GitHub;
+- a set of `.opam` files in [`opam-repository`][opam-repo] that point to this
+  release archive;
+- (optionally) a copy of these `.opam` files in the Tezos
+  [`opam-repository`][tezos-opam-repo].
+
+Most of this can be handled automatically by [`dune-release`][dune-release], as
+described in the instructions below.
+
+[git-tags]: https://git-scm.com/book/en/v2/Git-Basics-Tagging
+[pages-docs]: https://mirage.github.io/irmin
+[dune-release]: https://github.com/ocamllabs/dune-release
+[opam-repo]: https://github.com/ocaml/opam-repository
+
+## Releasing to opam-repository and GitHub
+
+- Check that no `.opam` files contain `pin-depends` fields. If so, release those
+  packages first.
```

Comment:
What's your workflow of releasing those packages first? Do you run `dune-release -p <pkg_name(s)>`? If so, might it be worth mentioning that?

Reply:
The pin-depended packages are usually https://github.com/mirage/index/ and https://github.com/mirage/repr/, so I suppose the process is "recursively execute this process on those repositories".